### PR TITLE
[SYNPY-1184] Add `includeTypes` to `synapseutils.walk()`

### PR DESCRIPTION
**Pull request by BrunoGrandePhD**
A small but incredibly useful quality of life tweak. Can this be included in 2.5? Pretty please? 

In cases where a mix of non-folder entities are expected, this avoids the user from having to make individual requests for each Synapse ID to figure out the type.

### CODE REVIEW

**Comment by BrunoGrandePhD** [comment]
Confirming that this will be slated for v2.6 so we can write proper tests and documentation in addition to adding a CLI argument.

**Comment by thomasyu888** [comment]
Adding cli argument out of scope of this PR. open new JIRA for that.

File: tests/unit/synapseutils/unit_test_synapseutils_walk.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        {"id": "syn124", "concreteType": "org.sagebionetworks.repo.model.Folder", "name": "test_folder"}
+    ]
+    child_list = [
+        [{"id": "syn2222", "concreteType": "File", "name": "test_file"},
+         {"id": "syn124", "concreteType": "org.sagebionetworks.repo.model.Folder", "name": "test_folder"}],
+        [{"id": "syn22223", "conreteType": "File", "name": "test_file_2"}]
+    ]
+    expected = [
+        (('parent_folder', 'syn123'), [('test_folder', 'syn124')], [('test_file', 'syn2222')]),
+        (('parent_folder/test_folder', 'syn124'), [], [('test_file_2', 'syn22223')])
```
Comment (thomasyu888) [inline tests/unit/synapseutils/unit_test_synapseutils_walk.py:49]:
```suggestion
        ((os.path.join('parent_folder', 'test_folder'), 'syn124'), [], [('test_file_2', 'syn22223')])
```

File: tests/unit/synapseutils/unit_test_synapseutils_walk.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
+from unittest.mock import patch
```
Comment (thomasyu888) [inline tests/unit/synapseutils/unit_test_synapseutils_walk.py:1]:
```suggestion
import os
from unittest.mock import patch
```

File: tests/unit/synapseutils/unit_test_synapseutils_walk.py (hunk 1 above)
Comment (thomasyu888) [reply tests/unit/synapseutils/unit_test_synapseutils_walk.py:49]: Must use `os` package to account for windows environment.

**Review by thomasyu888 — APPROVED** [review]

**Pull request merged**